### PR TITLE
nautilus: mon/MonClient: tolerate a rotating key that is slightly out of date

### PR DESCRIPTION
**Pull request by idryomov**
backport tracker: https://tracker.ceph.com/issues/50897

---

backport of https://github.com/ceph/ceph/pull/41368
parent tracker: https://tracker.ceph.com/issues/50390

### CODE REVIEW

**Review by ajarr — APPROVED** [review]

**Comment by ajarr** [comment]
@yuriw still need approval from the core team before merging this PR

**Review by ideepika — APPROVED** [review]
Looks good!

**Pull request merged**
